File: metadata/src/application/dtos/node_data_dto.rs
use super::PropertiesDataDto;
use crate::domain::models::{CreateNodeData, NodeData, NodeDataId};
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Debug, Clone, Deserialize, ToSchema)]
pub struct CreateNodeDataDto {
    pub key: String,
    pub properties: PropertiesDataDto,
}

impl CreateNodeDataDto {
    pub fn into_domain(self) -> CreateNodeData {
        CreateNodeData {
            key: self.key,
            properties: self.properties.into(),
        }
    }
}

#[derive(Debug, Serialize, ToSchema)]
pub struct NodeDataDto {
    pub node_data_id: NodeDataId,
    pub key: String,
    pub properties: PropertiesDataDto,
}

impl From<NodeData> for NodeDataDto {
    fn from(node_data: NodeData) -> Self {
        Self {
            node_data_id: node_data.node_data_id,
            key: node_data.key,
            properties: node_data.properties.into(),
        }
    }
}

File: web-ui/lib/api/dtos/node-data-dto.ts
import * as v from "valibot";
import { PropertiesDataDto } from "./property-data-dto";

export const CreateNodeDataDto = v.object({
  node_data_id: v.string(),
  key: v.string(),
  properties: PropertiesDataDto
});

export const NodeDataDto = v.object({
  node_data_id: v.string(),
  key: v.string(),
  properties: PropertiesDataDto
});
